### PR TITLE
Overlay map

### PR DESCRIPTION
**Pull request by FilippoLeon**
So, I worked a bit on a overlay for the map:
https://github.com/TeamPorcupine/ProjectPorcupine/issues/295

Especially for developement, but also for end-user.

Still a bit wonky, but should be useful for devs.

Currently I have maps for room id, power and oxygen.
Configurable trough lua.

Unfortunately I had to tweak the unity scene.

Comments appreciated.

### CODE REVIEW

**Comment by WardBenjamin** [comment]
Looks fantastic!

**Comment by TheRiven** [comment]
Like the power overlay! But just to let you know the `isPowerGenerator` check is being removed  (assuming my update gets approved) in favour of a positive for generators or negative for consumers `powerValue`

**Comment by FilippoLeon** [comment]
Good, I was just about to ask why we had that flag.

PS: if somebody has ideas for other overlays, I'll be happy to see if I can work on it.

**Comment by satch3l** [comment]
Any screenshots?

**Comment by bjubes** [comment]
@satch3l  there is one in the Issue request:
screenshot-from-295

**Comment by FilippoLeon** [comment]
If you scroll down there is the updated system:
https://github.com/TeamPorcupine/ProjectPorcupine/issues/295

File: Assets/Scripts/UI/Overlay/OverlayMap.cs [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+
+    /// <summary>
+    /// Script with user-defined lua valueAt functions
+    /// </summary>
+    Script script;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public string currentOverlay;
```
Comment (Grenkin1988) [inline Assets/Scripts/UI/Overlay/OverlayMap.cs]: Correct if I am wrong. There should be no public fields in class. Only public properties. (Encapsulation)

Comment (FilippoLeon) [inline Assets/Scripts/UI/Overlay/OverlayMap.cs]: If it's not a convention that the project uses (tell me if so), then no. By all means: properties are quite nice, but I wouldn't abuse them.

**Comment by henninglive** [comment]
Properties and public members should use PascalCasing.

File: Assets/Scripts/UI/Overlay/OverlayMap.cs (hunk 1 above)
Comment (Grenkin1988) [inline Assets/Scripts/UI/Overlay/OverlayMap.cs]: As I understand regarding public fields is just some kind Unity weird thing.

**Comment by quill18** [comment]
Overlays are good! Looking forward to see how much we can do with this.

**Pull request merged**